### PR TITLE
Add dependency duration standard metric

### PR DESCRIPTION
**Pull request by vishweshbankwar**

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzMonListExtensions.cs [hunk 1]
```diff
@@ -351,5 +352,31 @@ internal static string GetHttpDependencyName(this AzMonList tagObjects, string h
...
+                    {
+                        return AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeRpcSystem)?.ToString();
+                    }
+                case OperationType.Messaging:
+                    {
+                        return AzMonList.GetTagValue(ref tagObjects, SemanticConventions.AttributeMessagingSystem)?.ToString();
+                    }
+            }
+
+            return "Unknown";
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/AzMonListExtensions.cs:379]: This Unknown is not spec. Need to follow up on this with ingestion

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/StandardMetricsExtractionProcessor.cs [hunk 2]
```diff
@@ -71,6 +82,31 @@ private void ReportRequestDurationMetric(Activity activity, string statusCodeAtt
             _requestDuration.Record(activity.Duration.TotalMilliseconds, tags);
         }
 
+        private void ReportDependencyDurationMetric(Activity activity)
+        {
+            var monitorTags = TraceHelper.EnumerateActivityTags(activity);
+
+            var dependencyTarget = monitorTags.MappedTags.GetDependencyTarget(monitorTags.activityType);
```
Comment (rajkumar-rangaraj) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/StandardMetricsExtractionProcessor.cs:89]: In a follow up PR we might need to reconsider combining `GetDependencyTarget` and `GetDependencyType` and return `Dependency` objects with parameter `Target` and `Type`.

Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/StandardMetricsExtractionProcessor.cs:89]: tracking all follow ups here https://github.com/Azure/azure-sdk-for-net/issues/34012

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/StandardMetricsExtractionProcessor.cs [hunk 3]
```diff
@@ -71,6 +82,31 @@ private void ReportRequestDurationMetric(Activity activity, string statusCodeAtt
...
+            var statusCode = AzMonList.GetTagValue(ref monitorTags.MappedTags, SemanticConventions.AttributeHttpStatusCode);
+
+            var dependencyType = monitorTags.MappedTags.GetDependencyType(monitorTags.activityType);
+
+            TagList tags = default;
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.DependencyTargetKey, dependencyTarget));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.DependencyResultCodeKey, statusCode));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.MetricIdKey, StandardMetricConstants.DependencyDurationMetricIdValue));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.IsAutoCollectedKey, "True"));
+            tags.Add(new KeyValuePair<string, object>(StandardMetricConstants.IsSyntheticKey, "False"));
```
[the remark this answers is not in the document]
Comment (vishweshbankwar) [reply sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/StandardMetricsExtractionProcessor.cs:100]: Added to #34012

**Pull request merged**